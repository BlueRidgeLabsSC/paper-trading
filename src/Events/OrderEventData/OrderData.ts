import IData from "../IData";

export default class OrderData implements IData {
  price: number;
  quantity: number;
  type: string;
  cumulative_quantity: number;
  average_price: number;
  fees: number;
  symbol: string;
}
