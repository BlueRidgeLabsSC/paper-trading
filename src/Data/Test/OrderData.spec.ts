import OrderData from "../OrderData";

describe("test event initialization", () => {
  const data = new OrderData();
  data.price = 1.2;
  data.average_price = 1.6;
  data.cumulative_quantity = 30;
  data.fees = 20;
  data.symbol = "AAPL";
  data.type = "limit";

  test("expect object to match snapshot", () => {
    expect(data).toMatchSnapshot();
  });
});
