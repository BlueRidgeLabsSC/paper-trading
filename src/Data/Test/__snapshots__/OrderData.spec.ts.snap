// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test event initialization expect object to match snapshot 1`] = `
OrderData {
  "average_price": 1.6,
  "cumulative_quantity": 30,
  "fees": 20,
  "price": 1.2,
  "symbol": "AAPL",
  "type": "limit",
}
`;
